.aegir.js

```diff
@@ -3,12 +3,23 @@
 const IPFSFactory = require('ipfsd-ctl')
 const parallel = require('async/parallel')
 const MockPreloadNode = require('./test/utils/mock-preload-node')
+const EchoHttpServer = require('interface-ipfs-core/src/utils/echo-http-server')
 
 const ipfsdServer = IPFSFactory.createServer()
 const preloadNode = MockPreloadNode.createNode()
+const httpEchoServer = EchoHttpServer.createServer() // used by addFromURL
+
+const batch = (call, done, ...srvs) => parallel(srvs.map(srv => cb => {
+  if (srv === ipfsdServer) {
+    srv[call]()
+    cb()
```

Comment:
Wait for ipfsd server to start/stop before calling callback?

Reply:
done: https://github.com/ipfs/js-ipfs/pull/2304/commits/e7534ff624cf7171d1e1b352dc2eb6bf01b3acc7